### PR TITLE
[DRG] Some cleanup and simplification of job_util

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Originally intended to adress comment in PR #4415
- Changes method used for effect flag
- Changes raw item ID with enum
- Simplifies logic of spirit link, which had some questionable choices. Left TODO.
- General styling

## Steps to test these changes

Review by commit.
No changes should be observed.

### CODE REVIEW

File: scripts/globals/job_utils/dragoon.lua [hunk 1]
```diff
@@ -395,47 +395,47 @@ xi.job_utils.dragoon.useSpiritLink = function(player, target, ability)
...
-            if skin == drainamount then
-                player:delStatusEffect(xi.effect.STONESKIN)
-            else
-                local effect = player:getStatusEffect(xi.effect.STONESKIN)
-                effect:setPower(effect:getPower() - drainamount) -- fixes the status effect so when it ends it uses the new power instead of old
-                player:delMod(xi.mod.STONESKIN, drainamount) -- removes the amount from the mod
-            end
+        stoneskinPower = player:getMod(xi.mod.STONESKIN)
+
+        -- Stonesking is more powerfull than the amount to be drained.
```
Comment (WinterSolstice8) [inline scripts/globals/job_utils/dragoon.lua]: ` -- If stoneskin is more powerful than the amount to be drained.`

Comment (Xaver-DaRed) [reply scripts/globals/job_utils/dragoon.lua]: Me and my awesome English, am I right?

Comment (WinterSolstice8) [reply scripts/globals/job_utils/dragoon.lua]: I noticed the typo first, to be fair

File: scripts/globals/job_utils/dragoon.lua [hunk 2]
```diff
@@ -308,18 +309,18 @@ local function checkForRemovableEffectsOnSpiritLink(player, wyvern)
         -- Remove 2 erasable effects or effects that can be removed by -na
         local additionalRemovableEffects =
```
Comment (claywar) [inline scripts/globals/job_utils/dragoon.lua:310]: This could be simplified a bit and use set{}

Comment (WinterSolstice8) [reply scripts/globals/job_utils/dragoon.lua:310]: never seen that syntax before. neat!

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**